arch/x86/x86_intrins.h

```diff
@@ -84,4 +84,9 @@ static inline __m512i _mm512_zextsi128_si512(__m128i a) {
 #endif // __AVX512F__
 #endif // defined(_MSC_VER) && _MSC_VER < 1914
 
+/* Visual C++ toolchains before v142 have constant overflow in AVX512 intrinsics */
+#if defined(_MSC_VER) && defined(__AVX512F__) && !defined(_MM_K0_REG8)
```

Comment:
Is this the most reliable preprocessor definition for this (_MM_K0_REG8)?  Shouldn't we use an MSC_VER comparison?

Reply:
`_MSC_VER` doesn't show the toolchain or SDK version, only the IDE version... In this case it was using Visual C++ 2019 frontend, but the toolchain was from Visual C++ 2015 (v140) or Visual C++ 2017 (v141), so the used SDK (headers) was broken.

Reply:
Yeah and given the context above I say this should be fine.  Basically, they don't have the 8 bit mask registers defined.

Reply:
I really didn't want to diff all the available SDK versions to see if there is a header that defines the SDK version and which SDK version did fix each of the broken intrinsic macros... Making such complex tests would have warranted extensive commenting and might have slowed down preprocessing and compiling the code by quite a lot.